src/draw/nema_gfx/lv_nema_gfx_path.c

```diff
@@ -0,0 +1,151 @@
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice (including the next paragraph)
+ * shall be included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+/**
+ * @file lv_nema_gfx_path.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "../../core/lv_refr.h"
+
+#if LV_USE_NEMA_GFX
+#if LV_USE_NEMA_VG
+
+#include "lv_nema_gfx_path.h"
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+static int data_point = 0;
+static int seg_point  = 0;
+
+lv_nema_gfx_path_t * lv_nema_gfx_path_create(void)
+{
+    LV_PROFILER_BEGIN;
+    lv_nema_gfx_path_t * nema_gfx_path = lv_malloc_zeroed(sizeof(lv_nema_gfx_path_t));
+    LV_ASSERT_MALLOC(nema_gfx_path);
+    nema_gfx_path->seg = NULL;
+    nema_gfx_path->data = NULL;
+    nema_gfx_path->seg_size = 0;
+    nema_gfx_path->data_size = 0;
+    data_point = 0;
+    seg_point  = 0;
+    LV_PROFILER_END;
+    return nema_gfx_path;
+}
+
+void lv_nema_gfx_path_alloc(lv_nema_gfx_path_t * nema_gfx_path)
+{
+    LV_PROFILER_BEGIN;
+    nema_gfx_path->path = nema_vg_path_create();
+    nema_gfx_path->paint = nema_vg_paint_create();
+    nema_gfx_path->data = (float *) lv_malloc(nema_gfx_path->data_size * sizeof(float));
+    LV_ASSERT_MALLOC(nema_gfx_path->data);
+    nema_gfx_path->seg = (uint8_t *) lv_malloc(nema_gfx_path->seg_size * sizeof(uint8_t));
+    LV_ASSERT_MALLOC(nema_gfx_path->seg);
+    LV_PROFILER_END;
+}
+
+void lv_nema_gfx_path_destroy(lv_nema_gfx_path_t * nema_gfx_path)
+{
+    LV_PROFILER_BEGIN;
+    LV_ASSERT_NULL(nema_gfx_path);
+
+    if(nema_gfx_path->path != NULL) {
+        nema_vg_path_destroy(nema_gfx_path->path);
+        nema_gfx_path->path = NULL;
+    }
+
+    if(nema_gfx_path->paint != NULL) {
+        nema_vg_paint_destroy(nema_gfx_path->paint);
+        nema_gfx_path->paint = NULL;
+    }
+
+    if(nema_gfx_path->data != NULL) {
+        lv_free(nema_gfx_path->data);
+        nema_gfx_path->data = NULL;
+    }
+    if(nema_gfx_path->seg != NULL) {
+        lv_free(nema_gfx_path->seg);
+        nema_gfx_path->seg = NULL;
+    }
+    lv_free(nema_gfx_path);
+    LV_PROFILER_END;
+}
+
+void lv_nema_gfx_path_move_to(lv_nema_gfx_path_t * path, float x, float y)
+{
+    LV_ASSERT_NULL(path);
+    path->seg[seg_point++] = NEMA_VG_PRIM_MOVE;
+    path->data[data_point++] = x;
+    path->data[data_point++] = y;
```

Comment:
Can we add an array out of bounds check? It is necessary from the point of view of project experience.